### DISCUSSION TITLE
linking against system openssl

### DISCUSSION THREAD

**Discussion by kranurag7** [moby/moby#49018] [General]
As of now, cloning the repo and and invoking `./hack/make.sh dynbinary` builds dynamically linked binaries. In the current context, the binaries only links dynamically to libc libraries.

```bash
$ ldd bundles/dynbinary-daemon/*
bundles/dynbinary-daemon/docker-proxy:
        linux-vdso.so.1 (0x00007ffe923ee000)
        libresolv.so.2 => /lib/libresolv.so.2 (0x000078654b2ce000)
        libc.so.6 => /lib/libc.so.6 (0x000078654b0d9000)
        /lib64/ld-linux-x86-64.so.2 (0x000078654b2e4000)
bundles/dynbinary-daemon/dockerd:
        linux-vdso.so.1 (0x00007fffb024c000)
        libresolv.so.2 => /lib/libresolv.so.2 (0x00007d04d05ee000)
        libc.so.6 => /lib/libc.so.6 (0x00007d04d03f9000)
        /lib64/ld-linux-x86-64.so.2 (0x00007d04d5e2d000)
```

I want to link dockerd against system openssl instead of vendoring openssl. This'll help on security site if there's some openssl vulnerability down the line. As of now, `openssl` seems to be vendored when invoking `./hack/make.sh dynbinary`. 

Is there a way to link the binaries against system openssl. 

I've tried the following:
- looking for build tags in the repo but wasn't able to find one.
- looked at how other packagers are packaging (alpine and archlinux) it and seems like none of them link against system openssl. 

I'll really appreciate some references if there's a way to link the executables against system openssl.

**Comment by cpuguy83** [comment]
I don't think the official go binaries support linking to openssl as go has its own crypto stack.
In recent versions of go is able to link to boringcrypto (`GOEXPERIMENT=boringcrypto`).

There are forks to link to other crypto backends.
